docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,984 @@
+            imgUrl
+            imgAlt
+            author {
+              id
+              name
+            }
+            status
+          }
+        }
+      `,
+    })
+    // highlight-end
+  }
+
+  // rest of code that creates nodes
+}
+```
+
+Now you can write a function to subscribe to the data updates (like when a post is updated), and handle the new data coming in with a switch statement.
+
+```javascript:title=source-plugin/gatsby-node.js
+exports.sourceNodes = async (
+  { actions, createContentDigest, createNodeId, getNodesByType },
+  pluginOptions
+) => {
+  const { createNode, touchNode, deleteNode } = actions
+
+  // touch nodes to ensure they aren't garbage collected
+  getNodesByType(POST_NODE_TYPE).forEach(node => touchNode({ nodeId: node.id }))
+  getNodesByType(AUTHOR_NODE_TYPE).forEach(node =>
+    touchNode({ nodeId: node.id })
+  )
+
+  if (pluginOptions.previewMode) {
+    console.log("Subscribing to content updates...")
+    const subscription = await client.subscribe({
+      query: gql`
+        subscription {
+          posts {
+            id
+            slug
+            description
+            imgUrl
+            imgAlt
+            author {
+              id
+              name
+            }
+            status
+          }
+        }
+      `,
+    })
+    // highlight-start
+    subscription.subscribe(({ data }) => {
+      console.log(`Subscription received:`)
+      console.log(data.posts)
+      data.posts.forEach(post => {
+        const nodeId = createNodeId(`${POST_NODE_TYPE}-${post.id}`)
+        switch (post.status) {
+          case "deleted":
+            deleteNode({
+              node: getNode(nodeId),
+            })
+            break
+          case "created":
+          case "updated":
+          default:
+            // created and updated can be handled by the same code path
+            // the post's id is presumed to stay constant (or can be inferred)
+            createNode({
+              ...post,
+              id: createNodeId(`${POST_NODE_TYPE}-${post.id}`),
+              parent: null,
+              children: [],
+              internal: {
+                type: POST_NODE_TYPE,
+                content: JSON.stringify(post),
+                contentDigest: createContentDigest(post),
+              },
+            })
+            break
+        }
+      })
+    })
+    // highlight-end
+  }
+
+  // rest of code that creates nodes
+}
+```
+
+Posts that are changed on the backend while Gatsby is running will be created if they are new or updated, and deleted if they were deleted on the backend.
+
+You can test that this is working by running the site again and updating one of the posts. When you run the site this time you should see the message logged in the console: `Subscribing to content updates...` that was added. Now, running an `updatePost` or `deletePost` mutation on the GraphQL server will send information to the subscription because it is now listening.
```

Comment:
```suggestion
You can test that this is working by running the site again and updating one of the posts. When you run the site this time you should see a message logged in the console: `Subscribing to content updates...`. Now, running an `updatePost` or `deletePost` mutation on the GraphQL server will send information to the subscription because it is now listening.
```